src/Groups/gsets.jl

```diff
@@ -413,30 +413,29 @@ function _orbit_special_GAP(Omega::GSetByElements{<:GAPGroup, S}, omega::S) wher
     return res
 end
 
-function orbit(Omega::GSetByElements{FinGenAbGroup}, omega::T) where T
+function orbit(Omega::GSetByElements{T, S}, omega::S) where {T<:Union{Group, FinGenAbGroup}, S}
     return orbit_via_Julia(Omega, omega)
 end
 
 # simpleminded alternative directly in Julia
-function orbit_via_Julia(Omega::GSet, omega)
-    acts = gens(acting_group(Omega))
-    orbarray = [omega]
-    orb = Set(orbarray)
-    fun = action_function(Omega)
-    for p in orbarray
-      for g in acts
-        img = fun(p, g)
-        if !(img in orb)
-          push!(orbarray, img)
-          push!(orb, img)
-        end
+function orbit_via_Julia(Omega::GSet{T,S}, omega::S) where {T,S}
+  acts = gens(acting_group(Omega))
+  orb = IndexedSet([omega])
+  fun = action_function(Omega)
+
+  for p in orb
+    for g in acts
+      img = fun(p, g)::S
+      if !(img in orb)
+        push!(orb, img)
       end
     end
+  end
 
-    res = as_gset(acting_group(Omega), action_function(Omega), orbarray)
-    # We know that this G-set is transitive.
-    set_attribute!(res, :orbits => [res])
-    return res
+  res = as_gset(acting_group(Omega), action_function(Omega), collect(orb))
```

Comment:
this `collect` does basically do a copy of `orb.data`. IMO it would make sense to have some cheap way getting the underlying vector of an IndexedSet (where one has to promise to not modify it).

does that sound sane @fingolfin ?

Edit: I have just seen that `collect` has a special dispatch here to do exactly what I want. I am not sure if I find it a bit misleading under that name as `collect` usually does return a copy, see e.g.
```julia
julia> x = [1,2,3]
3-element Vector{Int64}:
 1
 2
 3

julia> collect(x) === x
false
```

Reply:
I agree the `collect` method is problematic and should be removed.

As to "getting the underlying vector of an IndexedSet (where one has to promise to not modify it)." normally I'd say the "API" for that is .. `orb.data` ? Of course we could also add a `data` method which returns that, but I am not sure we gain much here?

That said, do we even need to `collect` at all here? Doesn't the following work?
```suggestion
  res = as_gset(acting_group(Omega), action_function(Omega), orb)
```

Reply:
> That said, do we even need to collect at all here? Doesn't the following work?

yes, this should work as the `seeds` just needs to be an iterable, and the GSet will do it's own `collect`/copy if it decides that this makes sense